examples/pytorch/graphsage/experimental/train_dist_unsupervised.py

```diff
@@ -0,0 +1,393 @@
+
+            # Aggregate gradients in multiple nodes.
+            optimizer.step()
+            update_t.append(time.time() - compute_end)
+
+            pos_edges = pos_graph.number_of_edges()
+            neg_edges = neg_graph.number_of_edges()
+
+            step_t = time.time() - start
+            step_time.append(step_t)
+            iter_tput.append(pos_edges / step_t)
+            num_seeds += pos_edges
+            if step % args.log_every == 0:
+                print('[{}] Epoch {:05d} | Step {:05d} | Loss {:.4f} | Speed (samples/sec) {:.4f} | time {:.3f} s' \
+                        '| sample {:.3f} | copy {:.3f} | forward {:.3f} | backward {:.3f} | update {:.3f}'.format(
+                    g.rank(), epoch, step, loss.item(), np.mean(iter_tput[3:]), np.sum(step_time[-args.log_every:]),
+                    np.sum(sample_t[-args.log_every:]), np.sum(feat_copy_t[-args.log_every:]), np.sum(forward_t[-args.log_every:]),
+                    np.sum(backward_t[-args.log_every:]), np.sum(update_t[-args.log_every:])))
+            start = time.time()
+
+        print('[{}]Epoch Time(s): {:.4f}, sample: {:.4f}, data copy: {:.4f}, forward: {:.4f}, backward: {:.4f}, update: {:.4f}, #seeds: {}, #inputs: {}'.format(
+            g.rank(), np.sum(step_time), np.sum(sample_t), np.sum(feat_copy_t), np.sum(forward_t), np.sum(backward_t), np.sum(update_t), num_seeds, num_inputs))
+        epoch += 1
+
+        if args.standalone:
+            pred = generate_emb(model,  g, g.ndata['features'], args.batch_size_eval, device)
+        else:
+            pred = generate_emb(model.module, g, g.ndata['features'], args.batch_size_eval, device)
+        if g.rank() == 0:
+            f1_micro_eval, f1_micro_test, f1_macro_eval, f1_macro_test = compute_acc(pred, labels, global_train_nid, global_valid_nid, global_test_nid)
+            print('f1 micro eval {:.4f} f1 micro test {:.4f}, f1 macro eval {:.4f} f1 macro test {:.4f}'.format(
+                f1_micro_eval, f1_micro_test, f1_macro_eval, f1_macro_test))
+
+        # sync for eval and test
+        if not args.standalone:
+            th.distributed.barrier()
+
+    if not args.standalone:
+        g._client.barrier()
+
+        # save features into file
+        if g.rank() == 0:
+            th.save(pred, 'emb.pt')
+
+        dgl.distributed.shutdown_servers()
+        dgl.distributed.finalize_client()
+    else:
+        feat = g.ndata['features']
+        th.save(pred, 'emb.pt')
+
+def main(args):
+    if not args.standalone:
+        th.distributed.init_process_group(backend='gloo')
+    g = dgl.distributed.DistGraph(args.ip_config, args.graph_name, conf_file=args.conf_path)
+    print('rank:', g.rank())
+    print('number of edges', g.number_of_edges())
+
+    train_eids = dgl.distributed.edge_split(th.arange(g.number_of_edges()), g.get_partition_book(), force_even=True)
+    train_nids = dgl.distributed.node_split(th.arange(g.number_of_nodes()), g.get_partition_book())
```

Comment:
this is incorrect. the input of `node_split` and `edge_split` should be a boolean mask array, instead of a Id vector. I'll update the API to support both.

Reply:
I c. I will use th.ones to make the mask.